### PR TITLE
Rename `num_domains` to `num_additional_domains` in setup_pool

### PR DESCRIPTION
**Pull request by Sudha247**
Mentions that we actually create a pool of `num_domains + 1` along with the parent domain.

### CODE REVIEW

File: lib/task.mli [hunk 1]
```diff
@@ -8,7 +8,8 @@ type pool
 (** Type of task pool *)
 
 val setup_pool : num_domains:int -> pool
-(** Sets up a task execution pool with [num_domains]. *)
+(** Sets up a task execution pool with [num_domains + 1] including the parent
```
Comment (kayceesrk) [inline lib/task.mli:11]: `parent` should be `current`.

**Comment by kayceesrk** [comment]
Would it be useful to change the `num_domains` parameter as `num_additional_domains` to make it explicit? This is a breaking change. @ctk21 what are your thoughts?

**Comment by ctk21** [comment]
I can see advantages to cleaning that up to `num_additional_domains` (or `num_workers`).

We will probably need to make a release of domainslib soon that switches us from using `critical_section` to `Mutex`/`Condition`, so getting the change in for that release might make sense.

**Comment by ctk21** [comment]
Looks good to me.

**Pull request merged**